src/lib/Serialize.hs

```diff
@@ -63,13 +63,15 @@ getDexString (DataCon _ _ _ 0 [_, xs]) = case tryParseStringContent xs of
     tryParseStringContent tabAtom  = do
       TabLam (TabLamExpr i body) <- return tabAtom
       Fin (IdxRepVal n) <- return $ binderType i
-      Block _ (Nest castDecl (Nest offDecl Empty)) loadExpr <- return body
+      Abs (Nest castDecl (Nest offDecl Empty)) loadExpr <-
+        return $ inlineLastDeclBlock body
       Let v (DeclBinding _ _ (Op (CastOp IdxRepTy (Var i')))) <- return castDecl
       guard $ binderName i == i'
       Let v1 (DeclBinding _ _ (Op (PtrOffset (Var ptrName) (Var v')))) <- return offDecl
       guard $ binderName v == v'
       Hof (RunIO (Lam (LamExpr iob iobody))) <- return loadExpr
-      HoistSuccess (Block _ Empty (Op (PtrLoad (Var v1')))) <- return $ hoist iob iobody
+      HoistSuccess (Abs Empty (Op (PtrLoad (Var v1')))) <-
+        return $ hoist iob $ inlineLastDeclBlock iobody
```

Comment:
Same here, you could just modify the pattern slightly.

Reply:
Done.